chapters/descriptor_buffer.adoc

```diff
@@ -0,0 +1,250 @@
+
+layout (set = 1, binding = 0) buffer D { uint d; };
+layout (set = 1, binding = 1) buffer E { uint e; };
+layout (set = 1, binding = 2) buffer F { uint f; };
+
+layout (set = 2, binding = 0) buffer G { uint g; } array[3];
+
+void main() {
+    a = 10;
+    b = 20;
+    c = 30;
+    d = 40;
+    e = 50;
+    f = 60;
+    array[0].g = 70;
+    array[1].g = 80;
+    array[2].g = 90;
+}
+----
+
+With this shader, we will then have 3 `VkDescriptorSetLayout` in a single pipeline that exactly matches the shader interface.
+
+[source,c++]
+----
+// Set 0 and 1
+VkDescriptorSetLayoutBinding bindings_a[3] {
+    { binding = 0, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 },
+    { binding = 1, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 },
+    { binding = 2, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 1 }
+}
+
+// Set 2
+VkDescriptorSetLayoutBinding bindings_b[1] {
+    { binding = 0, descriptorType = VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, descriptorCount = 3 }
+}
+
+VkDescriptorSetLayout ds_layout_0(bindings_a); // Set 0
+VkDescriptorSetLayout ds_layout_1(bindings_a); // Set 1
+VkDescriptorSetLayout ds_layout_2(bindings_b); // Set 2
+VkPipelineLayout pipeline_layot([ds_layout_0, ds_layout_1, ds_layout_2]);
+----
+
+== Query Descriptor Set Layout Sizes
+
+Now using the new `vkGetDescriptorSetLayoutSizeEXT` and `vkGetDescriptorSetLayoutBindingOffsetEXT` commands, we can get info from the driver what size it needs to properly use these `VkDescriptorSetLayout`.
+
+[NOTE]
+====
+Don't make the assumption that `binding 0` will be the lowest offset!
+====
+
+[source,c++]
+----
+// This could be done in an array where the set/binding are indexes used to get the size/offsets
+VkDeviceSize set_0_size, set_1_size, set_2_size;
+vkGetDescriptorSetLayoutSizeEXT(device, ds_layout_0, &set_0_size);
+vkGetDescriptorSetLayoutSizeEXT(device, ds_layout_1, &set_1_size);
+vkGetDescriptorSetLayoutSizeEXT(device, ds_layout_2, &set_2_size);
+
+VkDeviceSize binding_0_offset, binding_1_offset, binding_2_offset;
+vkGetDescriptorSetLayoutBindingOffsetEXT(device, ds_layout_0, 0, &binding_0_offset);
+vkGetDescriptorSetLayoutBindingOffsetEXT(device, ds_layout_0, 1, &binding_1_offset);
+vkGetDescriptorSetLayoutBindingOffsetEXT(device, ds_layout_0, 2, &binding_2_offset);
+
+// ...
+----
+
+== Creating Descriptor Buffers
+
+We will create a "special" `VkBuffer` with `VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT` that now makes it a "descriptor buffer" `VkBuffer`.
+
+These buffers can be large and they hold a "look up table" to your resources and samplers.
+
+[NOTE]
+====
+The minimum required limit for `VkPhysicalDeviceDescriptorBufferPropertiesEXT::descriptorBufferAddressSpaceSize` is 128MB but link:https://vulkan.gpuinfo.org/displayextensionproperty.php?extensionname=VK_EXT_descriptor_buffer&extensionproperty=descriptorBufferAddressSpaceSize&platform=all[many devices can support 4GB]
```

Comment:
If possible, try keeping them in the docs.vulkan.org site.  This will ensure that all browsers keep the site index in cache and keep the site fast/smooth.